src/hotspot/share/gc/shenandoah/shenandoahInitLogger.cpp

```diff
@@ -24,63 +24,33 @@
-  } else {
-    log_info(gc, init)("Young Heuristics: %s", heap->young_generation()->heuristics()->name());
-    log_info(gc, init)("Young Generation Soft Size: " SIZE_FORMAT "%s",
-                       byte_size_in_proper_unit(heap->young_generation()->soft_max_capacity()),
-                       proper_unit_for_byte_size(heap->young_generation()->soft_max_capacity()));
-    log_info(gc, init)("Young Generation Max: " SIZE_FORMAT "%s",
-                       byte_size_in_proper_unit(heap->young_generation()->max_capacity()),
-                       proper_unit_for_byte_size(heap->young_generation()->max_capacity()));
-    log_info(gc, init)("Old Heuristics: %s", heap->old_generation()->heuristics()->name());
-    log_info(gc, init)("Old Generation Soft Size: " SIZE_FORMAT "%s",
-                       byte_size_in_proper_unit(heap->old_generation()->soft_max_capacity()),
-                       proper_unit_for_byte_size(heap->old_generation()->soft_max_capacity()));
-    log_info(gc, init)("Old Generation Max: " SIZE_FORMAT "%s",
-                       byte_size_in_proper_unit(heap->old_generation()->max_capacity()),
-                       proper_unit_for_byte_size(heap->old_generation()->max_capacity()));
-  }
-
-
-
-  log_info(gc, init)("Heap Region Count: " SIZE_FORMAT,
-                     ShenandoahHeapRegion::region_count());
-
-  log_info(gc, init)("Heap Region Size: " SIZE_FORMAT "%s",
-                     byte_size_in_exact_unit(ShenandoahHeapRegion::region_size_bytes()),
-                     exact_unit_for_byte_size(ShenandoahHeapRegion::region_size_bytes()));
-
-  log_info(gc, init)("TLAB Size Max: " SIZE_FORMAT "%s",
-                     byte_size_in_exact_unit(ShenandoahHeapRegion::max_tlab_size_bytes()),
-                     exact_unit_for_byte_size(ShenandoahHeapRegion::max_tlab_size_bytes()));
-
-  log_info(gc, init)("Humongous Object Threshold: " SIZE_FORMAT "%s",
-          byte_size_in_exact_unit(ShenandoahHeapRegion::humongous_threshold_bytes()),
-          exact_unit_for_byte_size(ShenandoahHeapRegion::humongous_threshold_bytes()));
+  log_info(gc, init)("Heap Region Count: " SIZE_FORMAT, ShenandoahHeapRegion::region_count());
+  log_info(gc, init)("Heap Region Size: " PROPERFMT, PROPERFMTARGS(ShenandoahHeapRegion::region_size_bytes()));
+  log_info(gc, init)("TLAB Size Max: " PROPERFMT, PROPERFMTARGS(ShenandoahHeapRegion::max_tlab_size_bytes()));
+  log_info(gc, init)("Humongous Object Threshold: " PROPERFMT, PROPERFMTARGS(ShenandoahHeapRegion::humongous_threshold_bytes()));
 }
 
-void ShenandoahInitLogger::print() {
-  ShenandoahInitLogger init_log;
-  init_log.print_all();
+void ShenandoahInitLogger::print_gc_specific() {
+  GCInitLogger::print_gc_specific();
+
+  ShenandoahHeap* heap = ShenandoahHeap::heap();
+  log_info(gc, init)("Mode: %s", heap->mode()->name());
+  log_info(gc, init)("Heuristics: %s", heap->global_generation()->heuristics()->name());
```

Comment:
Note - we still reference the `global_generation` abstraction in this file and we won't have that upstream. I wonder if we should push that abstraction back into `ShenandoahHeap` so that this code in `ShenandoahInitLogger` could call `ShenandoahHeap::heuristics` directly, with an implementation like:
```C++
ShenandoahHeuristics* ShenandoahHeap::heuristics() {
  return global_generation()->heuristics();
}
```

Reply:
Yeah.  Your proposed change might make the abstraction a bit cleaner.